bentoml/service.py

```diff
@@ -199,16 +199,16 @@ def _validation(inf_tasks):
                     task.discard(http_status=400, err_msg=str(e))
 
         # extract args
-        user_args = self.input_adapter.extract_user_func_args(_validation(inf_tasks))
+        user_args = self.input_adapter.extract_user_func_args(valid_tasks(inf_tasks))
         contexts = tuple(t.context for t in inf_tasks if not t.is_discarded)
 
         # call user function
         user_return = self.user_func(*user_args, contexts=contexts)
 
         if (
-            isinstance(user_return, (tuple, list))
+            isinstance(user_return, (Sequence, Iterable))
             and user_return
-            and isinstance(user_return[0], InferenceResult)
+            and all(map(lambda x: isinstance(x, InferenceResult), user_return))
```

Comment:
What is the point of more type checking here?

Reply:
It takes more time to check each one, but it does not bring benefits.

Reply:
I will revert that change, I made it since checking only the first element makes my hair raise :sweat_smile: I suppose that since the `user_return` shouldn't be empty it is okay to assume that there is always one element in the array.

Reply:
The `and user_return` make sure it couldn't be an empty list.